src/dali/plugin/pair_converter/csv/kraken.py

```diff
@@ -365,23 +390,23 @@ def _retrieve_cached_bars(
                     with gopen(file_path, "rt") as file:
                         rows = reader(file)
                         for row in rows:
-                            if all_bars and int(row[self.__TIMESTAMP_INDEX]) >= duration_timestamp:
+                            if all_bars and int(float(row[self.__TIMESTAMP_INDEX])) >= duration_timestamp:
                                 result.append(
                                     HistoricalBar(
                                         duration=timedelta(minutes=int(_KRAKEN_TIME_GRANULARITY[retry_count])),
-                                        timestamp=datetime.fromtimestamp(int(row[self.__TIMESTAMP_INDEX]), timezone.utc),
+                                        timestamp=datetime.fromtimestamp(int(float(row[self.__TIMESTAMP_INDEX])), timezone.utc),
```

Comment:
Also, double cast. This is repeated several times in the following lines as well.

Reply:
Yeah, the data had decimals in the timestamp. I can try to work on this in another PR and clean it up there. I honestly just wanted to get to an MVP with this.